tchannel/tornado/connection.py

```diff
@@ -471,25 +470,21 @@ def serve(self, handler):
                 # TODO Send error frame back
                 logging.exception("Failed to process %s", repr(message))
 
-    def send_error(self, code, description, message_id):
+    def send_error(self, error):
         """Convenience method for writing Error frames up the wire.
 
-        :param code:
-            Error code
-        :param description:
-            Error description
-        :param message_id:
-            Message in response to which this error is being sent
+        :param error:
+            TChannel Error.
         """
-        if code not in ErrorMessage.ERROR_CODES.keys():
-            raise FatalProtocolError(code)
-
-        return self._write(
-            ErrorMessage(
-                code=code,
-                description=description,
-                id=message_id,
-            ),
+
+        error_message = build_raw_error_message(error)
+        write_future = self._write(error_message)
+        tornado.ioloop.IOLoop.current().add_future(
```

Comment:
If your puts time counting things into their event hook, will it results incorrect value?